internal/pkg/agent/application/actions/handlers/handler_action_diagnostics.go

```diff
@@ -246,14 +277,20 @@ func (h *Diagnostics) diagUnits(ctx context.Context) []client.DiagnosticUnitResu
 }
 
 // diagUnits gathers diagnostics from components.
-func (h *Diagnostics) diagComponents(ctx context.Context) []client.DiagnosticComponentResult {
+func (h *Diagnostics) diagComponents(ctx context.Context, action *fleetapi.ActionDiagnostics) []client.DiagnosticComponentResult {
 	cDiag := make([]client.DiagnosticComponentResult, 0)
 	h.log.Debug("Performing component diagnostics")
 	startTime := time.Now()
 	defer func() {
 		h.log.Debugf("Component diagnostics complete. Took: %s", time.Since(startTime))
 	}()
-	rr, err := h.diagProvider.PerformComponentDiagnostics(ctx, []cproto.AdditionalDiagnosticRequest{})
+	additionalMetrics := []cproto.AdditionalDiagnosticRequest{}
+	for _, metric := range action.AdditionalMetrics {
+		if metric == "CPU" {
+			additionalMetrics = append(additionalMetrics, cproto.AdditionalDiagnosticRequest_CPU)
```

Comment:
This will take 30s per component, are there any timeouts anywhere that need to account for this?

It looks like diagnostics are already in a goroutine so it doesn't block anything else:

https://github.com/elastic/elastic-agent/blob/8ea0d04402aa62cf745d39ff535bc4ce7f642e23/internal/pkg/agent/application/actions/handlers/handler_action_diagnostics.go#L80-L89

Reply:
From what I can tell, we don't have a timeout associated with the action dispatch or handler